zenodo/modules/deposit/static/templates/zenodo_deposit/list.html

```diff
@@ -63,13 +64,21 @@
         <span ng-show="f.errored" class="text-danger">Error</span>
         <span ng-show="f.completed" class="text-success"><i class="fa fa-check"></i></span>
       </td>
-      <td ng-show="f" class="text-center"><a href="#" ng-click="filesVM.remove(f)"><i class="fa fa-trash-o"></a></td>
+      <td class="text-center">
```

Comment:
You can try to add ``ng-show="f.completed"`` to this td element so that the button is only visible when the file has been uploaded to the server.